### PR TITLE
feat(HMS-3521): Add configuration to enable/disable RBAC middleware

### PR DESCRIPTION
**Pull request by avisiedo**
This change add the configuration and skeleton for the RBAC middleware and unit tests.

The default value at config will be enabled in the code, but will be disabled until the implementation is full finished at HMS-3528

Checked by:

- `APP_ENABLE_RBAC=false make test-smoke` # run successfully
- `APP_ENABLE_RBAC=true make test-smoke`  # Fail with 403
- `make test` # Run successfully

### CODE REVIEW

File: internal/infrastructure/router/public.go
Comment (tiran) [inline internal/infrastructure/router/public.go:106]:
```suggestion
	if config.Get().Application.EnableRBAC {
		rbacMiddleware = middleware.RBACWithConfig(
			&middleware.RBACConfig{
				Skipper: nil,
				// TODO HMS-3522
				PermissionMap: middleware.RBACMap{},
			},
		)
	} else {
		rbacMiddleware := middleware.DefaultNooperation
	}
```

File: internal/infrastructure/middleware/rbac.go
Comment (tiran) [inline internal/infrastructure/middleware/rbac.go:14]: Please document what the RBACMap contains. Is it `route -> method -> permission` or `method -> route -> permission`?

File: internal/config/config_test.go
Comment (tiran) [inline internal/config/config_test.go:23]: I suggest that you turn these `TODO` comments into a refactoring ticket.

**Review by tiran — CHANGES_REQUESTED** [review]
Suggestion to make the PR shorter:
- Set `DefaultEnableRBAC = false` with a comment that RBAC is currently disabled because the feature is under development
- Remove all other places that set `enable_rbac: false`

File: deployments/clowder.yaml
Comment (tiran) [inline deployments/clowder.yaml:195]: I think this should be set to `true` by default. The app-interface change will set it to `false` for stage and ephemeral. You also have to pass the parameter as env var to the container.

File: configs/config.ci.yaml
Comment (tiran) [inline configs/config.ci.yaml:84]: Let's include the ticket number, so it is easier to find the places that have to be touched later.

```suggestion
  # TODO remove override when HMS-3521 is implemented
```

File: configs/config.example.yaml
Comment (tiran) [inline configs/config.example.yaml:86]:
```suggestion
  # TODO remove override when HMS-3521 is implemented
```

**Review by tiran — CHANGES_REQUESTED** [review]

File: .golangci.yaml
Comment (frasertweedale) [inline .golangci.yaml:143]: If we disable it, why not just delete?  Or please put a comment explaining why it is commented :)

File: deployments/clowder.yaml
Comment (frasertweedale) [reply deployments/clowder.yaml:195]: or we just remove the definition (now or later) - absence shall imply "enabled".

**Review by frasertweedale — APPROVED** [review]
One very small nit.  Not a show-stopper.  Please consider it, if you decide to change it, that's fine.  If not, that's OK too.  Then you can merge this PR :)

File: deployments/clowder.yaml
Comment (avisiedo) [reply deployments/clowder.yaml:195]: Indeed, I recognize it is confusing; my idea was "enabled by default" when everything is finished, but at the same time, don't brake anything until it is ready to work, so:

- When nothing is specified, the default value in the code (configuration) is enabled.
- In the config.yaml it is set the value as `false`, so it does not impact to the developer in other areas.
- In the makefile it is defined optional to `false`, to make possible to override the value from the CLI `APP_ENABLE_RBAC=true make run`, or the opposite.
- In the `clowder.yaml`, I had to define the parameter to be used from app-interface; it is useful when deploying in ephemeral so nothing is broken in IQE.
- In the app-interface, I force to false (until HMS-3528 is merged that enable everything); probably this is where no change is necessary at all; but when I thought about it at the beginning it made sense.

File: .golangci.yaml
Comment (avisiedo) [reply .golangci.yaml:143]: Honestly I don't remember why I commented the line. I leave it to keep the list of the linters on the file, but the reference before the `enable:` section is enough.

I have removed the line.

File: deployments/clowder.yaml
Comment (avisiedo) [reply deployments/clowder.yaml:195]: my conclusion; if we set this `true`, we will brake the IQE tests until the middleware is implemented (bear in mind they deploy by `bonfire deploy [--frontends true] idmsvc` with no additional parameters (which is right; what is in the makefiles is to make the developers life easier so the bonfire options are automatically filled to get the current commit state deployed); IMHO I would leave this parameter as it is.

Another question is if the app-interface MR is necessary; because we are managing the dafault value in the openshift template until we get the middleware ready. I think it could be cancelled.

What do you think?

Comment (avisiedo) [reply deployments/clowder.yaml:195]: Another thought in relation with the app-interface: from a perspective of a gradual deployment (1) change the deployment parameter in ephemeral (or remove), and when the feature looks working correctly, (2) change the deployment parameter (or remove) in stage, and get everything working.

So `app-interface` MR is necessary too.

Comment (tiran) [reply deployments/clowder.yaml:195]: The app-interface MR has landed

**Pull request merged**
